core/src/main/java/org/apache/iceberg/avro/PruneColumns.java

```diff
@@ -81,13 +81,18 @@ public Schema record(Schema record, List<String> names, List<Schema> fields) {
 
       Schema fieldSchema = fields.get(field.pos());
       // All primitives are selected by selecting the field, but map and list
-      // types can be selected by projecting the keys, values, or elements.
+      // types can be selected by projecting the keys, values, or elements. Empty
+      // Structs can be selected by selecting the record itself instead of it's children.
       // This creates two conditions where the field should be selected: if the
       // id is selected or if the result of the field is non-null. The only
       // case where the converted field is non-null is when a map or list is
       // selected by lower IDs.
       if (selectedIds.contains(fieldId)) {
-        filteredFields.add(copyField(field, field.schema(), fieldId));
+        if (fieldSchema != null) {
+          filteredFields.add(copyField(field, fieldSchema, fieldId));
```

Comment:
I'm not sure that I understand the reason for this change. Is this implementing the same change as the previous PR, but in the Avro `PruneColumns`?

It looks like if a struct field is selected and a sub-field is selected, then the selection for the struct isn't a full selection. But if a sub-field is not selected then the selection for the struct is a full selection. That doesn't make sense to me.

Reply:
As I'm thinking about this more, I think that the behavior in this class should always match `project`. I doubt there's a case where we want select behavior, right? In that case, shouldn't the `else` case check whether the type is a record and create an empty record?